plugin/src/main/java/org/opensearch/ml/settings/MLCommonsSettings.java

```diff
@@ -87,4 +87,7 @@ private MLCommonsSettings() {}
             Setting.Property.NodeScope,
             Setting.Property.Dynamic
         );
+
+    public static final Setting<Boolean> ML_COMMONS_ALLOW_MODEL_URL = Setting
+        .boolSetting("plugins.ml_commons.allow_model_url", false, Setting.Property.NodeScope, Setting.Property.Dynamic);
```

Comment:
Need tech writer's suggestion about the setting name. 

Hi Nate, can you help review if this setting name looks good or not?

This one will control whether user can upload their own model by setting url paramter in upload API. For security concern, we only allow user upload our pre-trained model https://opensearch.org/docs/latest/ml-commons-plugin/pretrained-models/, by specifying model name/version/format
POST /_plugins/_ml/models/_upload
{
  "name": "huggingface/sentence-transformers/all-MiniLM-L12-v2",
  "version": "1.0.1",
  "model_format": "TORCH_SCRIPT"
}
[12:18](https://amzn-aws.slack.com/archives/D035DCFAJQ6/p1682623081091489)
If user upload model by specifying url like this , our code will throw error
POST /_plugins/_ml/models/_upload
{
  "name": "sentence-transformers/all-MiniLM-L6-v2",
  "version": "1.0.0",
  "description": "test model",
  "model_format": "TORCH_SCRIPT",
  "model_config": {
    "model_type": "bert",
    "embedding_dimension": 384,
    "framework_type": "sentence_transformers"
  },
  "url": "https://github.com/opensearch-project/ml-commons/raw/2.x/ml-algorithms/src/test/resources/org/opensearch/ml/engine/algorithms/text_embedding/all-MiniLM-L6-v2_torchscript_sentence-transformer.zip?raw=true"
}

Reply:
@Naarcha-AWS

Reply:
Pinged @Naarcha-AWS offline, suggest to change to `plugins.ml_commons.allow_registering_model_via_url`

Reply:
Good question. @rbhavna , can you add a new setting to control enable/disable the local file uploading ? We can set it as disabled by default. Open source user can enable it if they need.

Reply:
That will add another setting. I'm just wondering if adding more setting also confuse customers or not.

The purpose is to allow models from other source than opensearch traced models or not. In that case why not use the same settings? May be renaming the settings to: `allow_registering_custom_models`?

Reply:
yeah makes sense. Using the same setting in both register model and upload model meta should be fine I guess.